packages/okta-auth-js/lib/browser/browser.js

```diff
@@ -308,11 +308,7 @@ proto.signOut = async function (options) {
   if (!idToken) {
     return sdk.closeSession() // can throw if the user cannot be signed out
     .then(function() {
-      if (postLogoutRedirectUri === defaultUri) {
-        window.location.reload();
-      } else {
-        window.location.assign(postLogoutRedirectUri);
-      }
+      window.location.assign(postLogoutRedirectUri);
```

Comment:
Ideally, I think it should also accept a callback to do in-app redirect without reloading the page.

Reply:
This is a fallback scenario, designed to simulate the `postLogoutRedirectUri` as closely as possible while using the XHR method. It has issues, including that it will break if 3rd party cookies are blocked. If developers can ensure 3rd party cookies are enabled and want to use the XHR method and redirect manually they should call `closeSession` instead of `signOut`

Reply:
I was running into this block when session expired and idToken had been cleared, which make me feel a pure in app redirect should just work without reloading the page.

Reply:
As I recall, if window.location.assign() is called with the current url, then nothing happens. `window.location.reload` is being used to **force** an app reload. The reason is to emulate the behavior of `postLogoutRedirectUri` - a redirect to Okta and back to the app would naturally reload the app. We are in a fallback here and I did not want to have an edge case where the app does not reload (only if the idToken is missing).